azerothcore/azerothcore-wotlk
# fix(Core/Spec): Changing spec now removes auras of spells with rank > 1

## Changes Proposed:
-  When we unlearn a spell, also remove all auras related to it during spec change.

## Issues Addressed:

Closes https://github.com/azerothcore/azerothcore-wotlk/issues/4284

## Tests Performed:

- Compiled without error
- Ran server +10 minutes
- Changed spec with mage to check if ice barrier buff rank 8,7,6,4,3,2,1 is removed


## How to Test the Changes:

1. Create a mage and level to 80 
2. Learn Ice barrier talent
3. Learn Ice Barrier spells > rank 1
4. Learn Dual spec
5. Use ice barrier > rank 1
6. Change spec while buff is active and notice it is removed


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
